src/components/editor.rs

```diff
@@ -161,7 +174,7 @@ impl<'a> Component for Editor<'a> {
     Ok(None)
   }
 
-  fn update(&mut self, action: Action, app_state: &AppState) -> Result<Option<Action>> {
+  fn update(&mut self, action: Action, app_state: &AppState<'_, DB>) -> Result<Option<Action>> {
     match action {
```

Comment:
all these `MenuPreview` queries will need to be moved to their respective database drivers